app/uk/gov/hmrc/directdebitupdateemailbackend/controllers/SjController.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.directdebitupdateemailbackend.controllers
+
+import com.google.inject.{Inject, Singleton}
+import ddUpdateEmail.models.Origin
+import ddUpdateEmail.models.journey.{SjRequest, SjResponse}
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents, Request, Result}
+import uk.gov.hmrc.directdebitupdateemailbackend.models.StartResult
+import uk.gov.hmrc.directdebitupdateemailbackend.services.StartService
+import uk.gov.hmrc.internalauth.client.{BackendAuthComponents, IAAction, Predicate, Resource, ResourceLocation, ResourceType}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SjController @Inject() (
+    startService: StartService,
+    auth:         BackendAuthComponents,
+    cc:           ControllerComponents
+)(implicit ex: ExecutionContext) extends BackendController(cc) {
+
+  private def internalAuthPermission(origin: String): Predicate.Permission = Predicate.Permission(
```

Comment:
Not a comment on code, but I wouldn't mind a quick walkthrough of this internal auth stuff if you have a spare few mins. I think I understand what it's doing, and can always read the docs in the repo so don't worry if not.